Parse/ParseClientConfiguration.h

```diff
@@ -0,0 +1,123 @@
+@property (nullable, nonatomic, copy, readwrite) NSString *clientKey;
+
+/**
+ Whether or not to enable pinning in the SDK.
+
+ The default value is `false`.
+ */
+@property (nonatomic, assign, readwrite, getter=isLocalDatastoreEnabled) BOOL localDatastoreEnabled;
+
+/**
+ When set, enables data sharing with an application group identifier.
+
+ After enabling - Local Datastore, `PFUser.+currentUser`, `PFInstallation.+currentInstallation` and all eventually commands
+ are going to be available to every application/extension in a group that have the same Parse applicationId.
+ */
+@property (nullable, nonatomic, copy, readwrite) NSString *applicationGroupIdentifier;
+
+/**
+ When set, controls the bundle identifier of the parent bundle to connect to.
+
+ @warning This property should only be set from inside an extension environment.
+ */
+@property (nullable, nonatomic, copy, readwrite) NSString *containingApplicationBundleIdentifier;
+
+/**
+ The maximum number of retry attempts to make upon a failed network request.
+ */
+@property (nonatomic, assign, readwrite) NSUInteger networkRetryAttempts;
+
+@end
+
+/*!
+ The `ParseClientConfiguration` represents the local configuration of the SDK to connect to the server with.
+
+ These configurations can be stored, copied, and compared, but cannot be safely changed once the SDK is initialized.
+
+ Use this object to construct a configuration for the SDK in your application, and pass it to
+ `Parse.+initializeWithConfiguration:`.
+ */
+@interface ParseClientConfiguration : NSObject<NSCopying>
+
+/**
+ The Parse.com application id to configure the SDK with.
+ */
+@property (nullable, nonatomic, copy, readonly) NSString *applicationId;
+
+/**
+ The Parse.com client key to configure the SDK with.
+ */
+@property (nullable, nonatomic, copy, readonly) NSString *clientKey;
+
+/**
+ Whether or not to enable pinning in the SDK.
+
+ The default value is `false`.
+ */
+@property (nonatomic, assign, readonly, getter=isLocalDatastoreEnabled) BOOL localDatastoreEnabled;
+
+/**
+ When set, enables data sharing with an application group identifier.
+
+ After enabling - Local Datastore, `PFUser.+currentUser`, `PFInstallation.+currentInstallation` and all eventually
+ commands are going to be available to every application/extension in a group that have the same Parse applicationId.
+ */
+@property (nullable, nonatomic, copy, readonly) NSString *applicationGroupIdentifier;
+
+/**
+ When set, controls the bundle identifier of the parent bundle to connect to.
+
+ @warning This property should only be set from inside an extension environment.
+ */
+@property (nullable, nonatomic, copy, readonly) NSString *containingApplicationBundleIdentifier;
+
+/**
+ The maximum number of retry attempts to make upon a failed network request.
+ */
+@property (nonatomic, assign, readonly) NSUInteger networkRetryAttempts;
+
++ (instancetype)configurationWithBlock:(void(^)(id<ParseMutableClientConfiguration> configuration))configurationBlock;
```

Comment:
Please double check if it's exported to Swift interface with the intended initializer, and not `ParseClientConfiguration.configurationWithBlock { ... in ... }`